cli/azd/resources/scaffold/base/modules/set-redis-conn.bicep

```diff
@@ -0,0 +1,41 @@
+param name string
+param keyVaultName string
+param passwordSecretName string
+param urlSecretName string
+
+resource redisConn 'Microsoft.Cache/redis@2024-03-01' existing = {
+  name: name
+}
+
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' existing = {
+  name: keyVaultName
+}
+
+resource passwordSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: passwordSecretName
+  parent: keyVault
+  properties: {
+    contentType: 'string'
+    attributes: {
+      enabled: true
+      exp: 0
+      nbf: 0
+    }
+    value: redisConn.listKeys().primaryKey
+  }
+}
+
+resource urlSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: urlSecretName
+  parent: keyVault
+  properties: {
+    contentType: 'string'
+    attributes: {
+      enabled: true
+      exp: 0
+      nbf: 0
+    }
+    value: 'rediss://:${redisConn.listKeys().primaryKey}@${redisConn.properties.hostName}:${redisConn.properties.sslPort}'
+  }
+}
+
```

Comment:
can we enable the app the use managed identity instead of having to add the redis secrets in keyvault?

Reply:
I'll defer to @puicchan for the final call here.

From what I'm seeing is that managed identity for redis isn't trivial to integrate with the app code for high availability for production scenarios.

As the [footnote](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-azure-active-directory-for-authentication#best-practices-for-microsoft-entra-authentication) in the official doc describes, to connect successfully, and maintain connection, one has to periodically fetch tokens and re-update the redis client/or reissue `AUTH` commands to avoid stale tokens being used: [js](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/samples/AzureCacheForRedis/ioredis.md#authenticate-with-microsoft-entra-id-handle-reauthentication), [python](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/samples/azure-aad-auth-with-redis-py.md#authenticate-with-azure-ad-handle-re-authentication) examples.  One also has to handle error conditions during these auth failures and refresh the connection. It's unclear if the proposed code samples are safe for multi-threaded application.

I don't have experience managing live services that integrate with Redis this way, but I can foresee that it requires some proper guidance and thinking.

My preference would be to scope this out as feature work for enabling managed identities for all databases. 

I quickly surveyed [awesome-azd](https://azure.github.io/awesome-azd/?tags=rediscache) for redis usage, none of the current templates have managed identity enabled yet.

Reply:
@wbreza @puicchan 

I created an issue to track the [feature](https://github.com/Azure/azure-dev/issues/4481). From engineering perspective, there is enough chunk of items to look at before we can commit to this.